yt/data_objects/construction_data_containers.py

```diff
@@ -684,6 +687,46 @@ def __init__(
             + self.ds.domain_offset
         )
         self._setup_data_source()
+
+        # determine if we need to extrapolate data for non-periodic domains
+        for i in range(3):
+            if not self.ds.periodicity[i] and (
+                self.left_edge[i] < self.ds.domain_left_edge[i]
+                or self.right_edge[i] > self.ds.domain_right_edge[i]
+            ):
+                self.fix_nonperiodic[i] = True
+            if self.fix_nonperiodic[i]:
+                if self.left_edge[i] < self.ds.domain_left_edge[i]:
+                    self.lo_buffer[i] = np.rint(
+                        (self.ds.domain_left_edge[i] - self.left_edge[i]) / self.dds[i]
+                    )
+                if self.right_edge[i] > self.ds.domain_right_edge[i]:
+                    self.hi_buffer[i] = np.rint(
+                        (self.right_edge[i] - self.ds.domain_right_edge[i])
+                        / self.dds[i]
+                    )
+        if any(self.fix_nonperiodic):
+            if np.count_nonzero(self.fix_nonperiodic) == 1:
+                extrap_dims = (
+                    "xyz"[np.where(self.fix_nonperiodic)[0][0]] + " " + "dimension"
```

Comment:
be careful here that "xyz" only works for cartesian geometries. `self.ds.coordinates.axis_order` is the general iterable for looping over axes names

Reply:
I agree, "xyz" isn't general, but doesn't a covering grid assume Cartesian geometry? I was under this impression, for a number of reasons, such as the container fields being Cartesian: 
```   
     _container_fields = (
        ("index", "dx"),
        ("index", "dy"),
        ("index", "dz"),
        ("index", "x"),
        ("index", "y"),
        ("index", "z"),
    )
```
or the coarse grid size being
```
self.base_dds = self.ds.domain_width / self.ds.domain_dimensions
```
Happy to change this iterable though, just want to make sure I am not missing anything!

Reply:
Let me get back to you on this